### PR TITLE
CSS-294 11ty Configuration Added for new documentation site

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description
This PR contains the below changes and additions
 * Pug files has been rewritten to nunjucks
 * 11ty has been added as configuration to serve the new documentation website
 * 11ty config has been written at the root in config.js
 * additional build scripts has been written (for now) to build and test css and nunjucks files for dev site
 * Globals needed for the website to run has been written in config.js
 * npx dependency has been removed from the build:dev script
 * Commitlint updates has been hold off
 * eleventy.cjs has been added inside site workspace

build-css.js is kept with all the functionalities of tasks folder for testing in this PR but will be removed once gulp tasks are migrated to vanilla nodejs for all the component-builder and component-builder-simple

https://jira.corp.adobe.com/browse/CSS-294

## How to test this PR locally ?
At the root of the project run:

``` 
yarn dev:11ty
```

## How and where has this been tested?
 - Testing will be done once this is merged with rebuild work happening in feature so keeping this PR as draft
 - **How this was tested:** 
 - **Browser(s) and OS(s) this was tested with:** 
 - Chrome on Mac

## Screenshots

Screenshot 2023-01-10 at 4 51 42 PM

Screenshot 2023-01-25 at 4 13 09 PM
Screenshot 2023-01-25 at 4 13 20 PM
Screenshot 2023-01-25 at 4 13 27 PM

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [ ] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [ ] I have tested these changes in Windows High Contrast mode.

- [x] This pull request is ready to merge.

### CODE REVIEW

File: commitlint.config.cjs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
```
Comment (castastrophe) [inline commitlint.config.cjs:1]: Can you pull back this update? We'll work on the module/common js transition another time.

File: components/icon/gulpfile.js [hunk 2]
```diff
@@ -17,13 +17,12 @@ const replace = require('gulp-replace');
 const sort = require('gulp-sort');
 const svgcombiner = require('gulp-svgcombiner');
 const svgstore = require('gulp-svgstore');
-const del = require('del');
+const rimraf = require('rimraf')
```
Comment (castastrophe) [inline components/icon/gulpfile.js:20]: You should be able to revert the changes from del to rimraf. Del is being used throughout the project, no need to substitute if for rimraf at this time.

File: components/tokens/custom-spectrum/custom-large-vars.css [hunk 3]
```diff
@@ -18,4 +18,7 @@ governing permissions and limitations under the License.
   --spectrum-edge-to-visual-only-100: 9px;
   --spectrum-edge-to-visual-only-200: 13px;
   --spectrum-edge-to-visual-only-300: 16px;
+
```
Comment (castastrophe) [inline components/tokens/custom-spectrum/custom-large-vars.css:21]: Double check the tokens updates in here - it seems unrelated to 11ty.

File: config.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
```
Comment (castastrophe) [inline config.js:2]: This should stay inside the tokens package - it's a style-dictionary config that should only be leveraged from there.

File: config/license.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+/*
+Copyright <%= YEAR %> Adobe. All rights reserved.
```
Comment (castastrophe) [inline config/license.js:2]: We should be able to make this update separately - it's not necessary for the 11ty implementation.

File: gulpfile.js [hunk 6]
```diff
@@ -215,10 +216,10 @@ exports.devHeavy = gulp.series(
   exports.devHeavy
 );
 
-exports['watch-relaunch'] = function() {
-  process.env['BROWSERSYNC_OPEN'] = true;
-  exports.watch();
-}
+// exports['watch-relaunch'] = function() {
```
Comment (castastrophe) [inline gulpfile.js:219]: I'd like to see the gulpfiles untouched by this if we can.

File: package.json [hunk 7]
```diff
@@ -20,6 +21,12 @@
     "url": "https://github.com/adobe/spectrum-css/issues"
   },
   "scripts": {
+    "build:all": "run-p build:11ty build:css",
+    "dev:11ty": "yarn dev && yarn build:11ty && NODE_ENV=development BROWSERSYNC_OPEN=true npx @11ty/eleventy --config=./site/.eleventy.cjs --serve",
+    "build:11ty": "npx @11ty/eleventy --config=./site/.eleventy.cjs",
+    "build:css": "node ./tasks/build-css.js && pretty-quick --fix --pattern \"{components}/**/*.css\"",
+    "build:css:watch": "node ./tasks/watch-css.js",
+    "build:watch": "run-p build:css:watch",
```
Comment (castastrophe) [inline package.json:29]:
```suggestion
    "dev:docs": "NODE_ENV=development BROWSERSYNC_OPEN=true yarn workspace spectrum-site run build --serve",
    "build:docs": "yarn workspace spectrum-site run build",
```

The site folder should be treated as a workspace and have it's own package with commands sourced from there in order to encapsulate dependencies.

File: package.json [hunk 8]
```diff
@@ -37,7 +44,7 @@
     "build:site": "gulp -LL",
     "ci-all": "yarn build",
     "clean": "gulp clean",
-    "dev": "NODE_ENV=development BROWSERSYNC_OPEN=true gulp dev",
+    "dev": "gulp dev",
```
Comment (castastrophe) [inline package.json:47]:
```suggestion
    "dev": "yarn dev:docs",
```

**Review by castastrophe — COMMENTED** [review]
Initial code review pass! Foremost I'd like to see a package.json added to the site folder and the workspaces updated. Commands and dependencies for the 11ty migration should source from there. Once you've got those updates in place we can dig into the details on the templates! 🎉

File: commitlint.config.cjs (hunk 1 above)
Comment (Rajdeepc) [reply commitlint.config.cjs:1]: Done

File: components/icon/gulpfile.js (hunk 2 above)
Comment (Rajdeepc) [reply components/icon/gulpfile.js:20]: del package doesnt export a function named del as such as per the docs. I was getting an error on build on the main branch also. that's why I used rimraf which I saw someone has added in the package.json
Docs: https://www.npmjs.com/package/del

Can you please re run the build from the main branch and check?

File: components/tokens/custom-spectrum/custom-large-vars.css (hunk 3 above)
Comment (Rajdeepc) [reply components/tokens/custom-spectrum/custom-large-vars.css:21]: Yes this is not related to 11ty. I double checked from the main branch. There is a new line from this file itself.
The below is the same file from the main branch. 
Screenshot 2023-01-30 at 11 25 33 AM

File: config.js (hunk 4 above)
Comment (Rajdeepc) [reply config.js:2]: Done

File: config/license.js (hunk 5 above)
Comment (Rajdeepc) [reply config/license.js:2]: Done

File: gulpfile.js (hunk 6 above)
Comment (Rajdeepc) [reply gulpfile.js:219]: Done

File: package.json (hunk 8 above)
Comment (Rajdeepc) [reply package.json]: Done

File: package.json [hunk 9]
```diff
@@ -74,18 +81,28 @@
     "y18n": "^5.0.5"
   },
   "devDependencies": {
+    "@11ty/eleventy": "^1.0.2",
+    "@11ty/eleventy-plugin-syntaxhighlight": "^4.2.0",
```
[the remark this answers is not in the document]
Comment (Rajdeepc) [reply package.json:55]: Done

File: tasks/build-css.js [hunk 10]
```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env node
+
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
```
[the remark this answers is not in the document]
Comment (Rajdeepc) [reply tasks/build-css.js:7]: Done

File: package.json (hunk 7 above)
Comment (Rajdeepc) [reply package.json]: Done.
@castastrophe Created a new folder called content and moved the nunjucks and md files inside them. You might see some files deleted and added in the commits. :)

File: components/icon/gulpfile.js (hunk 2 above)
Comment (Rajdeepc) [reply components/icon/gulpfile.js]: Yes it was causing some failures due to the migration.
